connectors/src/connectors/index.ts

```diff
@@ -1,445 +1,107 @@
-  google_drive: getGoogleDriveConfig,
-  intercom: getIntercomConfig,
-  webcrawler: () => {
-    throw new Error("Not implemented");
-  },
-};
-
-export const SET_CONNECTOR_CONFIGURATION_BY_TYPE: ConnectorProviderUpdateConfigurationMapping =
-  {
-    webcrawler: (
-      connectorId: ModelId,
-      configuration: WebCrawlerConfigurationType
-    ) => setWebcrawlerConfiguration(connectorId, configuration),
-    intercom: () => {
-      throw new Error("Not implemented");
-    },
-    slack: () => {
-      throw new Error("Not implemented");
-    },
-    microsoft: () => {
-      throw new Error("Not implemented");
-    },
-    notion: () => {
-      throw new Error("Not implemented");
-    },
-    github: () => {
-      throw new Error("Not implemented");
-    },
-    google_drive: () => {
-      throw new Error("Not implemented");
-    },
-    confluence: () => {
-      throw new Error("Not implemented");
-    },
-  };
-
-export const GARBAGE_COLLECT_BY_TYPE: Record<
-  ConnectorProvider,
-  ConnectorGarbageCollector
-> = {
-  confluence: () => {
-    throw new Error("Not implemented");
-  },
-  slack: () => {
-    throw new Error("Not implemented");
-  },
-  microsoft: async () => {
-    throw new Error("Not implemented");
-  },
-  notion: async () => {
-    throw new Error("Not implemented");
-  },
-  github: async () => {
-    throw new Error("Not implemented");
-  },
-  google_drive: googleDriveGarbageCollect,
-  intercom: async () => {
-    throw new Error("Not implemented");
-  },
-  webcrawler: () => {
-    throw new Error("Not implemented");
-  },
-};
-
-// If the connector has webhooks: stop processing them.
-// If the connector has long-running workflows: stop them.
-// Exclude the connector from the prod checks.
-export const PAUSE_CONNECTOR_BY_TYPE: Record<
-  ConnectorProvider,
-  ConnectorPauser
-> = {
-  confluence: pauseConfluenceConnector,
-  slack: pauseSlackConnector,
-  microsoft: pauseMicrosoftConnector,
-  notion: pauseNotionConnector,
-  github: pauseGithubConnector,
-  google_drive: pauseGoogleDriveConnector,
-  intercom: pauseIntercomConnector,
-  webcrawler: pauseWebcrawlerConnector,
-};
-
-// If the connector has webhooks: resume processing them, and trigger a full sync.
-// If the connector has long-running workflows: resume them. If they support "partial resync" do that, otherwise trigger a full sync.
-export const UNPAUSE_CONNECTOR_BY_TYPE: Record<
-  ConnectorProvider,
-  ConnectorUnpauser
-> = {
-  confluence: unpauseConfluenceConnector,
-  slack: unpauseSlackConnector,
-  microsoft: unpauseMicrosoftConnector,
-  notion: unpauseNotionConnector,
-  github: unpauseGithubConnector,
-  google_drive: unpauseGoogleDriveConnector,
-  intercom: unpauseIntercomConnector,
-  webcrawler: unpauseWebcrawlerConnector,
-};
+  | {
+      connectorProvider: "webcrawler";
+      params: {
+        dataSourceConfig: DataSourceConfig;
+        connectionId: string;
+        configuration: WebCrawlerConfiguration;
```

Comment:
Using a templated type for `configuration`, we should be able to go with only one type here!